genisys/modules/preseed.py

```diff
@@ -4,51 +4,58 @@
 from genisys.modules.base import Module
 from genisys.config_parser import YAMLParser
 import sys
-FILENAME = "preseed.cfg"
 
+FILENAME = "preseed.cfg"
+CERTIFICATE_STORE_PATH = Path("/etc/genisys/ssl")
 class Preseed(Module):
     """Generates a Preseed file to be served over the network to a booting Debian system"""
     def __init__(self: Self, config: YAMLParser):
         self.config = {}
         self.config["network"] = config.get_section("Network")
         self.config["users"] = config.get_section("Users")
         self.config["apps"] = config.get_section("Applications")
-    # end __init__
 
     def install_location(self: Self) -> Path:
         """Places the Preseed file in the TFTP root"""
         ftp_dir = Path(self.config["network"]["tftp_directory"])
         return Path(ftp_dir, FILENAME)
-    # end install_location
 
     def generate(self: Self) -> str:
         """Renders the Jinja template with the prespecified configurations"""
         loader = jinja2.PackageLoader('genisys')
         jinja_env = jinja2.Environment(loader=loader)
         template = jinja_env.get_template("preseed.cfg.jinja2")
 
-        # convert booleans to lowercase to match preseed format
+        # Convert booleans to lowercase to match preseed format
         for key, value in self.config["users"].items():
             if isinstance(value, bool):
                 self.config["users"][key] = str(value).lower()
 
+                # Read SSH key files and store their contents as strings
+                ssh_keys_contents = []
+                ssh_keys_dir = Path(self.config["users"].get("ssh-keys-dir", "../ssh_keys"))
+                for ssh_key_file in self.config["users"].get("ssh-keys", []):
+                    ssh_key_path = ssh_keys_dir / ssh_key_file
+                    with open(ssh_key_path, 'r') as f:
+                        ssh_keys_contents.append(f.read())
 
-        return template.render(settings=self.config["users"], ftp=self.config["network"].get("ftp", {}))
-    # end generate
+                # Determine SSL certificate path
+                ssl_cert_path = Path(self.config["network"].get("server", {}).get("ssl", {}).get("cert",
+                                                                                                 CERTIFICATE_STORE_PATH / "cert.pem"))
 
-# end class Preseed
+                ssl_cert_content = ""
+                if ssl_cert_path.exists():
```

Comment:
There should be a few more conditions here to match with some of the assumptions that the server script makes. I attached a diagram that I hope explains it.
![SSL Copy drawio](https://github.com/xeluior/genisys/assets/44057501/9f13b96a-6666-4e68-9c3a-c8bd569b14ac)